packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Providers/ParameterProvider.cs

```diff
@@ -30,6 +30,10 @@ public ParameterProvider(InputModelProperty inputProperty)
             Name = inputProperty.Name.ToVariableName();
             Description = FormattableStringHelpers.FromString(inputProperty.Description);
             Type = CodeModelPlugin.Instance.TypeFactory.CreateCSharpType(inputProperty.Type);
+            if (inputProperty is { IsRequired: false } && inputProperty.Type is not InputNullableType && !Type.IsCollection)
+            {
+                Type = Type.WithNullable(true);
+            }
```

Comment:
in general, we should not duplicate the code here, but this is a "must-do" in the current implementation.
This will be solved in one of my PR: https://github.com/microsoft/typespec/pull/3754